### ISSUE TITLE
cider - Inappropriate call if not paired read

### ISSUE THREAD

**Issue by crcmt** [hartwigmedical/hmftools#379]
Hello,
 
I get 'Inappropriate call if not paired read'
error message when running 'cider' tool on a single-end BAM file.

```
22:17:25.359 INFO  [main] - found constant region gene: IGHD3OR15-3B, type: IGHD, location: chr15:21012559-21012589(-)
22:17:25.359 INFO  [main] - found constant region gene: IGHD2OR15-2B, type: IGHD, location: chr15:21015048-21015078(-)
22:17:25.359 INFO  [main] - found constant region gene: IGHD1OR15-1B, type: IGHD, location: chr15:21017800-21017816(-)
22:17:25.429 INFO  [main] - found 372 gene locations
22:17:25.439 DEBUG [main] - Processing 432 potential sites in bam /home/staff/crc/tmp/test_cider/sorted_align.bam
22:17:25.440 INFO  [main] - 2 bam record consumer threads started
22:17:25.441 DEBUG [worker-0] - bam record consumer thread start
22:17:25.441 DEBUG [worker-1] - bam record consumer thread start
22:17:25.454 DEBUG [main] - bam reader start
22:17:25.455 DEBUG [main] - querying genome region: GenomeRegionImpl{chromosome=chr2, start=89884164, end=89886193}
22:17:25.466 DEBUG [main] - querying genome region: GenomeRegionImpl{chromosome=chr2, start=88930672, end=88932701}
22:17:25.467 DEBUG [main] - querying genome region: GenomeRegionImpl{chromosome=chr22, start=22409224, end=22411253}
22:17:25.472 DEBUG [main] - querying genome region: GenomeRegionImpl{chromosome=chr14, start=105862818, end=105864847}
22:17:25.473 DEBUG [main] - querying genome region: GenomeRegionImpl{chromosome=chr7, start=142380218, end=142382247}
22:17:25.474 ERROR [worker-1] - [Thread[worker-1,5,main]]: uncaught exception: java.lang.IllegalStateException: Inappropriate call if not paired read
22:17:25.474 ERROR [worker-0] - [Thread[worker-0,5,main]]: uncaught exception: java.lang.IllegalStateException: Inappropriate call if not paired read
java.lang.IllegalStateException: Inappropriate call if not paired read
    at htsjdk.samtools.SAMRecord.requireReadPaired(SAMRecord.java:892)
    at htsjdk.samtools.SAMRecord.getFirstOfPairFlag(SAMRecord.java:950)
    at com.hartwig.hmftools.cider.CiderReadScreener.asyncProcessSamRecord(CiderReadScreener.kt:54)
    at com.hartwig.hmftools.cider.CiderApplication$readBamFile$asyncBamRecordHander$1.invoke(CiderApplication.kt:120)
    at com.hartwig.hmftools.cider.CiderApplication$readBamFile$asyncBamRecordHander$1.invoke(CiderApplication.kt:119)
    at com.hartwig.hmftools.cider.AsyncBamReader$BamRecordConsumerThread.run(AsyncBamReader.kt:165)
java.lang.IllegalStateException: Inappropriate call if not paired read
    at htsjdk.samtools.SAMRecord.requireReadPaired(SAMRecord.java:892)
    at htsjdk.samtools.SAMRecord.getFirstOfPairFlag(SAMRecord.java:950)
    at com.hartwig.hmftools.cider.CiderReadScreener.asyncProcessSamRecord(CiderReadScreener.kt:54)
    at com.hartwig.hmftools.cider.CiderApplication$readBamFile$asyncBamRecordHander$1.invoke(CiderApplication.kt:120)
    at com.hartwig.hmftools.cider.CiderApplication$readBamFile$asyncBamRecordHander$1.invoke(CiderApplication.kt:119)
    at com.hartwig.hmftools.cider.AsyncBamReader$BamRecordConsumerThread.run(AsyncBamReader.kt:165)

```

Does 'cider' need paired-end as input ?

Thanks

**Comment by crcmt** [comment]
Unfortunately, I cannot share the data as under controlled-access policy,

I will try to reproduce the error with less-sensitive data. 

Thanks!

**Comment by hongwingl** [comment]
This is fixed in v1.0.3 of cider
